modules/ismp/clients/tendermint/src/lib.rs

```diff
@@ -0,0 +1,191 @@
+			.map_err(|e| ismp::error::Error::Custom(e.to_string()))?;
+
+		let mut state_machine_map: BTreeMap<StateMachineId, Vec<StateCommitmentHeight>> =
+			BTreeMap::new();
+		let mut updated_consensus_state = consensus_state.clone();
+
+		let app_hash: [u8; 32] = consensus_proof
+			.signed_header
+			.header
+			.app_hash
+			.as_bytes()
+			.try_into()
+			.map_err(|_| Error::Custom("Invalid app hash length".to_string()))?;
+
+		let state_commitment = StateCommitmentHeight {
+			commitment: StateCommitment {
+				timestamp: updated_state.verified_timestamp,
+				overlay_root: None,
+				state_root: primitive_types::H256(app_hash),
+			},
+			height: updated_state.trusted_state.height,
+		};
+
+		state_machine_map.insert(
+			StateMachineId {
+				state_id: StateMachine::Evm(consensus_state.chain_id),
+				consensus_state_id,
+			},
+			vec![state_commitment],
+		);
+
+		updated_consensus_state.tendermint_state =
+			CodecTrustedState::from(&updated_state.trusted_state);
+
+		Ok((updated_consensus_state.encode(), state_machine_map))
+	}
+
+	fn verify_fraud_proof(
+		&self,
+		host: &dyn IsmpHost,
+		trusted_consensus_state: Vec<u8>,
+		proof_1: Vec<u8>,
+		proof_2: Vec<u8>,
+	) -> Result<(), Error> {
+		let update_1: TendermintConsensusUpdate =
+			Decode::decode(&mut &proof_1[..]).map_err(|e| Error::Custom(e.to_string()))?;
+		let update_2: TendermintConsensusUpdate =
+			Decode::decode(&mut &proof_2[..]).map_err(|e| Error::Custom(e.to_string()))?;
+
+		let consensus_state: ConsensusState = Decode::decode(&mut &trusted_consensus_state[..])
+			.map_err(|e| Error::Custom(e.to_string()))?;
+
+		let height_1 = update_1.tendermint_proof.signed_header.header.height;
+		let height_2 = update_2.tendermint_proof.signed_header.header.height;
+		if height_1 != height_2 {
+			return Err(Error::Custom(
+				"Fraud proofs must be for the same block height".to_string(),
+			));
+		}
+
+		if proof_1 == proof_2 {
+			return Err(Error::Custom("Fraud proofs are identical".to_string()));
+		}
+
+		let trusted_state: TrustedState = consensus_state.clone().tendermint_state.into();
+
+		let time = host.timestamp().as_secs();
+
+		let consensus_proof_1 = update_1
+			.tendermint_proof
+			.to_consensus_proof()
+			.map_err(|e| Error::Custom(e.to_string()))?;
+
+		let consensus_proof_2 = update_2
+			.tendermint_proof
+			.to_consensus_proof()
+			.map_err(|e| Error::Custom(e.to_string()))?;
+
+		verify_header_update(trusted_state.clone(), consensus_proof_1, time)
+			.map_err(|e| Error::Custom(e.to_string()))?;
+		verify_header_update(trusted_state, consensus_proof_2, time)
+			.map_err(|e| Error::Custom(e.to_string()))?;
+
+		Ok(())
+	}
+
+	fn consensus_client_id(&self) -> ConsensusClientId {
+		self.client_id
+	}
+
+	fn state_machine(&self, id: StateMachine) -> Result<Box<dyn StateMachineClient>, Error> {
+		match id {
+			StateMachine::Evm(_) => Ok(Box::new(TendermintEvmStateMachine::<H, T>::default())),
```

Comment:
Check if state machine is inside the SupportedStateMachines before returning  a value